packages/plugin-edgedb/src/types.ts

```diff
@@ -0,0 +1,389 @@
+import type { Client } from 'edgedb';
+import {
+  FieldMap,
+  FieldNullability,
+  InputFieldMap,
+  InputShapeFromFields,
+  InterfaceParam,
+  MaybePromise,
+  Normalize,
+  ObjectRef,
+  SchemaTypes,
+  ShapeWithNullability,
+  TypeParam,
+} from '@pothos/core';
+import type { EdgeDBObjectFieldBuilder } from './edgedb-field-builder';
+import { GraphQLResolveInfo } from 'graphql';
+
+export interface EdgeDBQueryBuilder {
+  default: unknown;
+}
+export interface EdgeDBDriver extends Client {}
+
+export type EdgeDBModels<
+  Name extends EdgeDBTypeKeys,
+  Types extends SchemaTypes = SchemaTypes,
+> = Types['EdgeDBTypes']['default'][Name] extends infer T ? T : never;
+
+export type EdgeDBTypeKeys<Types extends SchemaTypes = SchemaTypes> =
+  keyof Types['EdgeDBTypes']['default'] extends infer Key
+    ? Key extends string
+      ? Key
+      : never
+    : never;
+
+export type EdgeDBDefaultExportKeyTypes<DefaultExports> = {
+  [K in keyof DefaultExports]-?: K extends string
+    ? string
+    : K extends number
+    ? number
+    : K extends symbol
+    ? symbol
+    : never;
+}[keyof DefaultExports];
+
+type EdgeDBDefaultExportKey<
+  DefaultExports,
+  KeyType extends string | number | symbol = EdgeDBDefaultExportKeyTypes<DefaultExports>,
+> = Extract<keyof DefaultExports, KeyType>;
+
+export type EdgeDBModelTypes<Types extends SchemaTypes> =
+  Types['EdgeDBTypes']['default'] extends infer ObjectTypeMap
+    ? ObjectTypeMap extends object
+      ? ObjectTypeMap
+      : never
+    : never;
+
+export type EdgeDBModelShape<
```

Comment:
EdgeDB's Query Builder contains a default which includes all db schema's with their `properties` and `relations`, or named as `links` in edgeDB, of the schemas and can be used to recreate the `PrismaTypes` structure for type safety.

## Example of EdgeDBTypes
```ts
export interface types {
  "std": {
    ...
  };
  "schema": {
    ...
  };
  "cfg": {
    ...
  };
  "default": {
    "Comment": Comment;
    "Follow": Follow;
    "Media": Media;
    "Post": Post;
    "PostMedia": PostMedia;
    "Profile": Profile;
    "User": User;
  };
  "sys": {
     ....
  };
}

export interface Post extends std.$Object {
  "big_int_id": bigint;
  "content"?: string | null;
  "created_at": Date;
  "published": boolean;
  "title": string;
  "updated_at": Date;
  "author"?: User | null;
}
```